src/SprykerEco/Zed/Computop/Business/Payment/Handler/Saver/Init/SofortResponseSaver.php

```diff
@@ -41,10 +42,10 @@ function () use ($responseTransfer): void {
      */
     protected function savePaymentComputopEntity(ComputopSofortInitResponseTransfer $responseTransfer): void
     {
-        $paymentEntity = $this->getPaymentEntity();
-        $paymentEntity->setPayId($responseTransfer->getHeader()->getPayId());
-        $paymentEntity->setXId($responseTransfer->getHeader()->getXId());
-        $paymentEntity->save();
+        $this->getPaymentEntity()
+            ->setPayId($responseTransfer->getHeader()->getPayId())
```

Comment:
```suggestion
            ->setPayId($responseTransfer->getHeaderOrFail()->getPayId())
```

same for similar cases below

Reply:
still not fixed

Reply:
Fixed